### ISSUE TITLE
v9.4 Basement preprocessor fails with fatal error

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#8333]
Issue overview
--------------
Example file LgOffVAVusingBasement.idf fails with a fatal error when run the the basement.exe packaged with v9.4.0 (on Windows, maybe others?).
```
   ** Severe  ** Output:PreprocessorMessage="GroundTempCalc - Basement" has the following Severe conditions:
   **   ~~~   ** EPlusOutput: Calculations have resulted in invalid results (NAN). (Could be divide by zero.
   **   ~~~   ** Overflows. Other causes.) Check Input Parameters.  Hint: Start from defaults for each object
   **   ~~~   ** and work forward.
   ** Severe  ** Output:PreprocessorMessage="GroundTempCalc - Basement" has the following Fatal condition:
   **   ~~~   ** Program terminates due to preceding condition(s).
   **  Fatal  ** Preprocessor condition(s) cause termination.
```
The same file runs to completion using the v9.3.0 basement.exe  Did something change with the Fortran build options for v9.4?

### Workaround
Copy the v9.3 basement.exe file over the v9.4 basement.exe.  It may be found in EnergyPlusV9-3-0\PreProcess\GrndTempCalc\

### Details
Some additional details for this issue (if relevant):
 - Platform Windows 10
 - Version of EnergyPlus 9.4.0 release
 - Direct e-mail from PNNL

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by jmarrec** [comment]
I can't run that file but I'm not getting the same error on Unbut 18.04 with 9.4.0 official.

```shell
$ cp /usr/local/EnergyPlus-9-4-0/ExampleFiles/LgOffVAVusingBasement.idf .
   
$ energyplus -x -w /usr/local/EnergyPlus-9-4-0/WeatherData/USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw -d out-ori LgOffVAVusingBasement.idf 
ExpandObjects Started.
 Begin reading Energy+.idd file.
 Done reading Energy+.idd file.
ExpandObjects Finished. Time:     0.112
EnergyPlus Starting
EnergyPlus, Version 9.4.0-998c4b761e, YMD=2020.10.16 11:29
**FATAL:GetSurfaceData: Errors discovered, program terminates.
EnergyPlus Run Time=00hr 00min  0.45sec
Program terminated: EnergyPlus Terminated--Error(s) Detected.
(py38)julien@8333_Basement_v940_crash (master *%=)$ cat out-ori/eplusout.err 
Program Version,EnergyPlus, Version 9.4.0-998c4b761e, YMD=2020.10.16 11:29,
   ************* Beginning Zone Sizing Calculations
   ** Severe  ** BuildingSurface:Detailed="UWB", invalid Outside Boundary Condition Object="SURFPROPOTHSDCOEFBASEMENTAVGWALL".
   **   ~~~   **  no OtherSideCoefficients of that name.
   ** Severe  ** BuildingSurface:Detailed="UWR", invalid Outside Boundary Condition Object="SURFPROPOTHSDCOEFBASEMENTAVGWALL".
   **   ~~~   **  no OtherSideCoefficients of that name.
   ** Severe  ** BuildingSurface:Detailed="UWF", invalid Outside Boundary Condition Object="SURFPROPOTHSDCOEFBASEMENTAVGWALL".
   **   ~~~   **  no OtherSideCoefficients of that name.
   ** Severe  ** BuildingSurface:Detailed="UWL", invalid Outside Boundary Condition Object="SURFPROPOTHSDCOEFBASEMENTAVGWALL".
   **   ~~~   **  no OtherSideCoefficients of that name.
   ** Severe  ** BuildingSurface:Detailed="UF", invalid Outside Boundary Condition Object="SURFPROPOTHSDCOEFBASEMENTAVGFLOOR".
   **   ~~~   **  no OtherSideCoefficients of that name.
   **  Fatal  ** GetSurfaceData: Errors discovered, program terminates.
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=5
   ..... Last severe error=BuildingSurface:Detailed="UF", invalid Outside Boundary Condition Object="SURFPROPOTHSDCOEFBASEMENTAVGFLOOR".
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 5 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 0 Warning; 5 Severe Errors; Elapsed Time=00hr 00min  0.45sec
```

**Comment by mjwitte** [comment]
It looks like the basement preprocessor didn't run at all. There should be messages from the preprocessor to std out.

**Comment by jmarrec** [comment]
Seems like there is no option to run basement / slab from the CLI. https://unmethours.com/question/29916/issues-running-energy-plus-through-windows-command-prompt-with-groundheattransfer-objects/

Should we open an enhancement request to add that capability or something? runenergyplus/ RunEPlus.bat is marked deprecated as of August 2015. Or rather than opening an issue, re-title https://github.com/NREL/EnergyPlus/issues/7097

**Comment by jmarrec** [comment]
## TL;DR

**I ran it successfully on both Ubuntu 18.04 and Windows 10, with E+ 9.4.0 official.**



## Ubuntu Testing

So, I've used `runenergyplus` on Ubuntu and it works

```
export ENERGYPLUS_DIR=/usr/local/EnergyPlus-9-4-0
/usr/local/EnergyPlus-9-4-0/runenergyplus LgOffVAVusingBasement.idf USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw
```

basement ran fine, took 10 years to converge
```
[...]
 Year:           9 Day:         365 Converge Status: F
 3-D Solution Has Not Converged, Delta T>0.1
 Max DeltaT=8.169E-002
 Elapsed time: 759.16 seconds
 Year:          10 Day:          30 Converge Status: T
 Year:          10 Day:          60 Converge Status: T
 Year:          10 Day:          90 Converge Status: T
 Year:          10 Day:         120 Converge Status: T
 Year:          10 Day:         150 Converge Status: T
 Year:          10 Day:         180 Converge Status: T
 Year:          10 Day:         210 Converge Status: T
 Year:          10 Day:         240 Converge Status: T
 Year:          10 Day:         270 Converge Status: T
 Year:          10 Day:         300 Converge Status: T
 Year:          10 Day:         330 Converge Status: T
 Year:          10 Day:         360 Converge Status: T
 Year:          10 Day:         365 Converge Status: T
 3-D Solution Has Converged
 Elapsed time: 844.59 seconds
 IYR=          10   Program Terminated
STOP GroundTempCalc Completed Successfully
```

Then energyplus worked as well:

```
(py38)julien@8333_Basement_v940_crash (master *+%=)$ cat Output/LgOffVAVusingBasement.err 
Program Version,EnergyPlus, Version 9.4.0-998c4b761e, YMD=2020.10.19 12:56,
   ** Warning ** Output:PreprocessorMessage="ExpandObjects" has the following Warning condition:
   **   ~~~   ** Cannot find Energy+.idd as specified in Energy+.ini.
   ** Warning ** Output:PreprocessorMessage="ExpandObjects" has the following Warning conditions:
   **   ~~~   ** Since the Energy+.IDD file cannot be read no range or choice checking was
   **   ~~~   ** performed.
   ** Warning ** Output:PreprocessorMessage="ExpandObjects" has the following Warning conditions:
   **   ~~~   ** The Site:GroundTemperature:BuildingSurface object present in the file will
   **   ~~~   ** not be used. Intead the values from Slab will be used.
   ************* Beginning Zone Sizing Calculations
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-2T" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-3T" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-4T" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="COR-T" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-1F" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-2F" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-3F" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="COR-F" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-1I" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-2I" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-3I" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ** Warning ** SetUpZoneSizingArrays: Sizing for Zone="PER-4I" will use Sizing:Zone specifications listed for Zone="PER-1T".
   ************* Beginning System Sizing Calculations
   ************* Beginning Plant Sizing Calculations
   ************* Testing Individual Branch Integrity
   ************* All Branches passed integrity testing
   ************* Testing Individual Supply Air Path Integrity
   ************* All Supply Air Paths passed integrity testing
   ************* Testing Individual Return Air Path Integrity
   ************* All Return Air Paths passed integrity testing
   ************* No node connection errors were found.
   ************* Beginning Simulation
   ************* Simulation Error Summary *************
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 12 Warning; 0 Severe Errors.
   ************* EnergyPlus Completed Successfully-- 15 Warning; 0 Severe Errors; Elapsed Time=00hr 00min  1.27sec
```

## Windows testing

I used EP-launch with 9.4.0 official, LgOffVAVusingBasement.idf, weather `USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw`

I got exactly the same as on Ubuntu. Basement convergence after 10 days, and E+ simulated successfully.

**Comment by mjwitte** [comment]
@jmarrec Interesting. Here's what I see with 9.4.0 (64-bit) on Windows 10.
```
 1-D solution has not converged, Delta T>0.05
 1-D solution has converged, MaxDeltaT=   0.0000000000000000
 Year:           1 Day:          30 Converge Status: F
 Year:           1 Day:          60 Converge Status: F
 Year:           1 Day:          90 Converge Status: F
 Year:           1 Day:         120 Converge Status: F
 Year:           1 Day:         150 Converge Status: F
 Year:           1 Day:         180 Converge Status: F
 Year:           1 Day:         210 Converge Status: F
 Year:           1 Day:         240 Converge Status: F
 Year:           1 Day:         270 Converge Status: F
 Year:           1 Day:         300 Converge Status: F
 Year:           1 Day:         330 Converge Status: F
 Year:           1 Day:         360 Converge Status: F
 Year:           1 Day:         365 Converge Status: F
 3-D Solution Has Not Converged, Delta T>0.1
 Max DeltaT=-99999.000
 Elapsed time: 65.13 seconds
Note: The following floating-point exceptions are signalling: IEEE_INVALID_FLAG IEEE_DIVIDE_BY_ZERO IEEE_DENORMAL
STOP GroundTempCalc Terminated--Error(s) Detected.
```

**Comment by cskarupa** [comment]
I am getting this same error - was anything ever figured out?

**Comment by mjwitte** [comment]
Not yet, use the workaround:
Copy the v9.3 basement.exe file over the v9.x basement.exe. It may be found in EnergyPlusV9-3-0\PreProcess\GrndTempCalc\

**Comment by mjwitte** [comment]
Well, actually, it depends on which error you are referring to and which platform.

**Comment by cskarupa** [comment]
Yeah I think that's what I'm going to do. I'm referring to the same issue as your first post - running 9.4 and 9.5 produces same fatal error on Windows. But runs with 9.3 Basement.exe file. Thanks!
